packages/aws-cdk/test/integ/cli/aws-helpers.ts

```diff
@@ -0,0 +1,143 @@
+import * as AWS from 'aws-sdk';
+import { log } from './cdk-helpers';
+
+interface Env {
+  account: string;
+  region: string;
+}
+
+export let testEnv = async (): Promise<Env> => {
+  const response = await new AWS.STS().getCallerIdentity().promise();
+
+  const ret: Env = {
+    account: response.Account!,
+    region: process.env.AWS_REGION ?? process.env.AWS_DEFAULT_REGION ?? 'us-east-1',
+  };
+
+  testEnv = () => Promise.resolve(ret);
+  return ret;
+};
+
+export let cloudFormation = makeAwsCaller(AWS.CloudFormation);
+export let sns = makeAwsCaller(AWS.SNS);
+export let iam = makeAwsCaller(AWS.IAM);
+export let lambda = makeAwsCaller(AWS.Lambda);
+export let sts = makeAwsCaller(AWS.STS);
+
+/**
+ * Perform an AWS call from nothing
+ *
+ * Create the correct client, do the call and resole the promise().
+ */
+async function awsCall<
+  A extends AWS.Service,
+  B extends keyof ServiceCalls<A>,
+>(ctor: new (config: any) => A, call: B, request: First<ServiceCalls<A>[B]>): Promise<Second<ServiceCalls<A>[B]>> {
+  const env = await testEnv();
+  const cfn = new ctor({ region: env.region });
+  const response = cfn[call](request);
+  try {
+    return await response.promise();
+  } catch (e) {
+    const newErr = new Error(`${call}(${JSON.stringify(request)}): ${e.message}`);
+    (newErr as any).code = e.code;
+    throw newErr;
```

Comment:
Maybe also chain the `cause` somehow so the original trace is not forever lost?

Reply:
I'm doing this in particular because the traces are useless (async and all).

Also, it's tests.